include/TrustWalletCore/TWCardano.h

```diff
@@ -0,0 +1,22 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "TWBase.h"
+#include "TWData.h"
+
+TW_EXTERN_C_BEGIN
+
+TW_EXPORT_STRUCT
+struct TWCardano;
+
+// The minimum ADA amount needed for a UTXO.  See https://docs.cardano.org/native-tokens/minimum-ada-value-requirement
+// Input is serialized TokenBundle protobuf object.
+TW_EXPORT_STATIC_METHOD
+uint64_t TWCardanoMinAdaAmount(TWData *_Nonnull tokenBundle) TW_VISIBILITY_DEFAULT;
```

Comment:
can we hide this logic somehow? the usage and api don't look good

Reply:
App needs this calculation _before_ putting together SigningInput, because in case of token transfer, it has to know how much ADA to transfer alongside of the token.  I imagine:
1. app puts together output tokenBundle, containing the token type and amount requested to be transferred
2 app invokes MinAdaAmount with tokenBundle
3. app creates SigningInput, with requested ADA amount calculated above, plus tokenBundle prepared

Reply:
If you look at token support on other chain, this is complicated as hell, coin transfer takes 2 steps, and token transfer needs 3 steps?

Reply:
This is Cardano, and token transfer in a UTXO chain is an interesting beast...
It is very unfortunate that token-only UTXOs are not possible, a minimum amount of ADA must be also there.
App can hardcode to sg like 2 ADA, but it will not be optimal.

Reply:
Let's use hardcode value to simplify this?

Reply:
This is optional, app can use hardcoded value or use this method.